### PR TITLE
Feature: Add iron's RefinedTypeOps support

### PR DESCRIPTION
**Pull request by vbergeron**
Using the Iron library, a common pattern is to hide the constrained type behind an opaque alias, to disallow similarly constrained types (The same result can be obtained using `DescribedAs`, but I prefer not messing description with identity).

```scala
type RefinedIntConstraint = Interval.ClosedOpen[0, 10]
opaque type RefinedInt <: Int = Int :| RefinedIntConstraint

object RefinedInt extends RefinedTypeOpsImpl[Int, RefinedIntConstraint, RefinedInt]
```

Once the alias is opaque, it is not possible anymore to derive instances with the `Value :| Predicate` type pattern. 

This PR provides instances relying on the given `RefinedTypeOps.Mirror[T]` instance that is provided by `RefinedTypeOpsImpl`.

I wonder if the provided tests are enough or if there is corner cases I need to be aware of.

### CODE REVIEW

File: integrations/iron/src/test/scala-3/sttp/iron/codec/iron/TapirCodecIronTestScala3.scala [hunk 1]
```diff
@@ -18,6 +18,11 @@ import io.github.iltotore.iron.constraint.all.*
 import sttp.tapir.Validator
 import sttp.tapir.ValidationError
 
+// Opaque aliases can't be included into class bodies
+type RefinedIntConstraint = Interval.ClosedOpen[0, 10]
+opaque type RefinedInt <: Int = Int :| RefinedIntConstraint
```
[the remark this answers is not in the document]
Comment (vbergeron) [reply integrations/iron/src/test/scala-3/sttp/iron/codec/iron/TapirCodecIronTestScala3.scala]: Right, I moved the opaque definition to its own file.
